### PR TITLE
Remove TOML annotation for secrets in test config and load them from env vars

### PR DESCRIPTION
**Pull request by lukaszcl**
This allows to load E2E test secrets from env vars and default ~/.testsecrets file. It also adds `ghsecrets` CLI tool to easily set the secrets as Github Secrets.

This PR is required for TT-1381

---
**Below is a summarization created by an LLM (gpt-4-0125-preview). Be mindful of hallucinations and verify accuracy.**

## Why
The changes introduce functionalities to improve the handling of secrets for test environments, primarily focusing on enhancing security and flexibility in managing environment-specific configurations. These updates allow for a more structured approach to injecting secrets into test configurations, which is crucial for maintaining the integrity and security of sensitive data in automated testing environments.

## What
- **config/README.md**: Updated the documentation to reflect the new approach for managing test secrets, moving from a TOML config-based secret management to an environment variable-based strategy. This includes detailed instructions on how to set up and use various methods to inject secrets into test environments.
- **config/testconfig.go**: Introduced significant enhancements to load test configuration values from environment variables. This includes loading various types of secrets such as Chainlink images, wallet keys, RPC URLs, and more into the `TestConfig` struct. Also, added a method to load secrets from a file, enhancing flexibility in secret management.
- **config/testconfig_test.go**: Added new unit tests to ensure the correct loading of configuration values from environment variables into the `TestConfig` struct.
- **go.mod** and **go.sum**: Included `github.com/joho/godotenv` as a dependency to facilitate the loading of environment variables from files, supporting the new secrets management approach.
- **k8s/config/overrides.go**: Added a new constant to prefix environment variables specifically for E2E tests, enhancing the clarity and organization of environment variables.
- **k8s/environment/runner.go**: Updated to forward environment variables with the new E2E test prefix, facilitating better separation and management of test-specific configurations.
- **tools/ghsecrets**: Added a new tool for managing GitHub secrets, including a Makefile, README, and the necessary Go modules. This tool aids in setting test secrets in GitHub actions, supporting the secure management of sensitive information in CI/CD pipelines.

### CODE REVIEW

File: config/testconfig.go [hunk 1]
```diff
@@ -39,3 +48,232 @@ type TestConfig struct {
...
+			envVars[key] = append(envVars[key], value)
+		}
+	}
+
+	if len(envVars) == 0 {
+		logger.Debug().Msgf("Not setting test config key '%s' because environment variables that match '%s' regex not found", key, regexpStr)
+		return nil
+	}
+
+	for network, keys := range envVars {
```
Comment (Tofel) [inline config/testconfig.go:209]: nit: should the map key be called `network`? maybe it should be something general? that's at least what I assume based on function name, that it can read any env var groups, where the key doesn't need to represent a network

File: config/testconfig.go [hunk 2]
```diff
@@ -39,3 +48,232 @@ type TestConfig struct {
...
+		keyPath := fmt.Sprintf("%s.%s", key, network)
+		viperLock.Lock()
+		viper.Set(keyPath, keys)
+		viperLock.Unlock()
```
Comment (Tofel) [inline config/testconfig.go:213]: why do we need to use mutex here? in what situation would we experience concurrent operations on the same test config?

Comment (lukaszcl) [reply config/testconfig.go]: @Tofel `viper` does not handle synchonization well. There are multiple issues about it. I just pushed another commit replacing viper - we don't need this new module

File: config/testconfig.go (hunk 1 above)
Comment (lukaszcl) [reply config/testconfig.go]: @Tofel this is no longer relevant. I updated the code (because [of this](https://github.com/smartcontractkit/chainlink-testing-framework/pull/1017#discussion_r1666641626)) and added tests. Please check again

File: config/testconfig.go [hunk 3]
```diff
@@ -39,3 +49,380 @@ type TestConfig struct {
...
+			group := matches[1] // Use the first capture group for grouping
+			groupedVars[group] = append(groupedVars[group], value)
+			logger.Debug().Msgf("Will override test config from env var '%s'", key)
+		}
+	}
+	return groupedVars
+}
+
+func loadEnvVarSingleMap(pattern string) map[string]string {
+	logger := logging.GetTestLogger(nil)
```
Comment (skudasov) [inline config/testconfig.go:296]: Do you know if we need a test logger here or if we don't? I am not sure it can properly work with `nil`
Also, please check other places.

File: config/testconfig.go [hunk 4]
```diff
@@ -39,3 +49,380 @@ type TestConfig struct {
 	Seth                   *seth.Config           `toml:"Seth"`
 	NodeConfig             *NodeConfig            `toml:"NodeConfig"`
 }
+
+// Read config values from environment variables
+func (c *TestConfig) ReadConfigValuesFromEnvVars() error {
```
Comment (skudasov) [inline config/testconfig.go:54]: It'd be good to reflect this function in documentation so people understand they need to add their new env vars there.

File: config/testconfig.go [hunk 5]
```diff
@@ -39,3 +49,380 @@ type TestConfig struct {
...
+		}
+	}
+	return groupedVars
+}
+
+func loadEnvVarSingleMap(pattern string) map[string]string {
+	logger := logging.GetTestLogger(nil)
+	re := regexp.MustCompile(pattern)
+	singleVars := make(map[string]string)
+	for _, env := range os.Environ() {
```
Comment (skudasov) [inline config/testconfig.go:299]: Maybe we can use `loadEnvVarGroupedMap` everywhere? Looks like a common case that can cover `loadEnvVarSingleMap` too.

**Review by skudasov — APPROVED** [review]

File: config/testconfig.go (hunk 3 above)
Comment (lukaszcl) [reply config/testconfig.go:296]: @skudasov yeah, we need it to show logs about overrides. Also, it works fine with testing.T nil

File: config/testconfig.go (hunk 5 above)
Comment (lukaszcl) [reply config/testconfig.go:299]: @skudasov I thought about it but we can't. These are different functions with different return values. It's better to split them

File: config/testconfig.go (hunk 4 above)
Comment (lukaszcl) [reply config/testconfig.go:54]: @skudasov QA will not have to add env vars here unless they add a secret.

File: config/README.md [hunk 6]
```diff
@@ -24,9 +24,69 @@ It's up to the user to provide a way to read the config from file and unmarshal
...
+| Grafana Base URL              | `E2E_TEST_GRAFANA_BASE_URL`                                         | `E2E_TEST_GRAFANA_BASE_URL=base_url`                |
+| Grafana Dashboard URL         | `E2E_TEST_GRAFANA_DASHBOARD_URL`                                    | `E2E_TEST_GRAFANA_DASHBOARD_URL=url`                |
+| Grafana Bearer Token          | `E2E_TEST_GRAFANA_BEARER_TOKEN`                                     | `E2E_TEST_GRAFANA_BEARER_TOKEN=token`               |
+| Pyroscope Server URL          | `E2E_TEST_PYROSCOPE_SERVER_URL`                                     | `E2E_TEST_PYROSCOPE_SERVER_URL=url`                 |
+| Pyroscope Key                 | `E2E_TEST_PYROSCOPE_KEY`                                            | `E2E_TEST_PYROSCOPE_KEY=key`                        |
+| Pyroscope Environment         | `E2E_TEST_PYROSCOPE_ENVIRONMENT`                                    | `E2E_TEST_PYROSCOPE_ENVIRONMENT=env`                |
+| Pyroscope Enabled             | `E2E_TEST_PYROSCOPE_ENABLED`                                        | `E2E_TEST_PYROSCOPE_ENABLED=true`                   |
```
Comment (Tofel) [inline config/README.md:60]: imho neither `E2E_TEST_PYROSCOPE_ENVIRONMENT` nor `E2E_TEST_PYROSCOPE_ENABLED` are secret. Environment is just some random user-provided identifier, enabled is a `bool`

File: config/README.md [hunk 7]
```diff
@@ -24,9 +24,69 @@ It's up to the user to provide a way to read the config from file and unmarshal
...
+| RPC WebSocket URL per network | `E2E_TEST_(.+)_RPC_WS_URL` or `E2E_TEST_(.+)_RPC_WS_URL_(\d+)$`     | `E2E_TEST_ARBITRUM_RPC_WS_URL=ws_url`               |
+| Loki Tenant ID                | `E2E_TEST_LOKI_TENANT_ID`                                           | `E2E_TEST_LOKI_TENANT_ID=tenant_id`                 |
+| Loki Endpoint                 | `E2E_TEST_LOKI_ENDPOINT`                                            | `E2E_TEST_LOKI_ENDPOINT=url`                        |
+| Loki Basic Auth               | `E2E_TEST_LOKI_BASIC_AUTH`                                          | `E2E_TEST_LOKI_BASIC_AUTH=token`                    |
+| Loki Bearer Token             | `E2E_TEST_LOKI_BEARER_TOKEN`                                        | `E2E_TEST_LOKI_BEARER_TOKEN=token`                  |
+| Grafana Base URL              | `E2E_TEST_GRAFANA_BASE_URL`                                         | `E2E_TEST_GRAFANA_BASE_URL=base_url`                |
+| Grafana Dashboard URL         | `E2E_TEST_GRAFANA_DASHBOARD_URL`                                    | `E2E_TEST_GRAFANA_DASHBOARD_URL=url`                |
```
Comment (Tofel) [inline config/README.md:55]: don't think this one is a secret

File: tools/ghsecrets/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  echo $(go env GOPATH)/bin
+  ```
+
+  This command will print the path where Go binaries are installed, typically something like /home/username/go/bin
+
+- Add the following line at the end of the file:
+
+  ```sh
+  export PATH="$PATH:<path-to-go-bin>"
+  ```
```
Comment (Tofel) [inline tools/ghsecrets/README.md:43]: pls update this part with `asdf` in mind, exporting `GOPATH` like that when you're using `asdf` might mess up your env

Comment (lukaszcl) [reply tools/ghsecrets/README.md:43]: mentioned asdf in https://github.com/smartcontractkit/chainlink-testing-framework/pull/1020

File: config/README.md (hunk 7 above)
Comment (lukaszcl) [reply config/README.md:55]: @Tofel thanks, removed in https://github.com/smartcontractkit/chainlink-testing-framework/pull/1020/commits/5ec97034df6500b796a7a8a29436aebe119a1c0c

File: config/README.md (hunk 6 above)
Comment (lukaszcl) [reply config/README.md:60]: @Tofel thanks, removed in https://github.com/smartcontractkit/chainlink-testing-framework/pull/1020/commits/5ec97034df6500b796a7a8a29436aebe119a1c0c

File: config/testconfig.go (hunk 4 above)
Comment (lukaszcl) [reply config/testconfig.go:54]: added doc in https://github.com/smartcontractkit/chainlink-testing-framework/pull/1020
